### PR TITLE
19346 Update alternate_names and other tables + misc script updates

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#19346

*Description of changes:*

* Addition of following columns to alternate_names/alternate_names_history tables
  * business_start_date
  * dissolution_date
  * state
  * state_filing_id
  * admin_freeze
  * last_modified
* Addition of required FKs and indexes to  alternate_names/alternate_names_history tables
*  Add alternate_name_id column to following tables and corresponding FK entry
  * comments
  * request_tracker
  * filings
  * dc_connections
  * dc_issued_business_user_credentials
  * documents/document_versions
 * Misc cleanup of temp functions + extensions created by migrations scripts

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql [hunk 1]
```diff
@@ -1498,6 +1498,10 @@ DROP FUNCTION has_non_legal_name_change;
 DROP FUNCTION update_filing_json_party_ids;
 DROP FUNCTION rename_jsonb_key;
 DROP FUNCTION update_legal_name;
+DROP FUNCTION cast_le_to_leh;
+DROP FUNCTION get_previous_le_history_entry;
+DROP FUNCTION insert_into_leh;
+DROP EXTENSION hstore;
```
Comment (argush3) [inline legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql:1504]: misc cleanup to remove temp functions in this migration script

File: legal-api/migrations/versions/004f7558063f_second.py [hunk 2]
```diff
@@ -653,12 +686,17 @@ def upgrade():
...
+    sa.Column('legal_entity_id', sa.Integer(), nullable=True),  # Allow null values
+    sa.Column('alternate_name_id', sa.Integer(), nullable=True),  # Allow null values
     sa.ForeignKeyConstraint(['legal_entity_id'], ['legal_entities.id']),
+    sa.ForeignKeyConstraint(['alternate_name_id'], ['alternate_names.id']),
     sa.ForeignKeyConstraint(['user_id'], ['users.id']),
     sa.PrimaryKeyConstraint('id')
     )
 
+    # Add a check constraint to ensure at least one of legal_entity_id or alternate_name_id is non-null
+    op.execute("ALTER TABLE dc_issued_business_user_credentials ADD CONSTRAINT at_least_one_non_null CHECK (legal_entity_id IS NOT NULL OR alternate_name_id IS NOT NULL)")
```
Comment (argush3) [inline legal-api/migrations/versions/004f7558063f_second.py:698]: `legal_entity_id` was previous non-nullable but with addition of `alternate_name_id`, made this field nullable.

Added a constraint to ensure at least `legal_entity_id` or `alternate_name` is non-null

**Comment by argush3** [comment]
linting errors will be addressed in #19549

**Review by kzdev420 — APPROVED** [review]
LGTM

**Review by JazzarKarim — APPROVED** [review]
LGTM!

**Pull request merged**